cmake/patch_llvm.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python3
+
+"""
+Script to apply a set of patches to llvm-project sources.
+"""
+
+import argparse
+import os
+import subprocess
+import sys
+
+
+def main():
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument(
+        "patchdir",
+        help="Set of patches to apply. This should be a directory containing one or more ordered *.patch files.",
```

Comment:
The code as currently written requires this to be provided as an absolute path. We should either ease this requirement, or document it here. 

I'm considering the scenario where the user wants to run this script to apply patches manually. The scenario as described in CMakeLists.txt has them in the root of LLVM-embedded-toolchain-for-Arm.

Reply:
I can see how the paths can get muddled since the git commands need to run in the llvm-project location, whereas the patch paths will be specified relative to the script. So I've adjusted the git arguments to use `-C` and convert the patch paths to absolute automatically.